images/metering-ansible-operator/roles/meteringconfig/library/k8s_status.py

```diff
@@ -328,14 +328,25 @@ def merge_status(self, old, new):
         merged = copy.deepcopy(old_conditions)
 
         for condition in new_conditions:
-            idx = self.get_condition_idx(merged, condition['type'])
-            if idx is not None:
-                merged[idx] = condition
-            else:
-                merged.append(condition)
+          new_merged_list = self.filter_existing_conditions(merged, condition['type'])
+          new_merged_list.append(condition)
+
+          merged = new_merged_list
+
         new['conditions'] = merged
         return new
 
+    def filter_existing_conditions(self, conditions, type):
+        new_conditions = []
+
+        for _, condition in enumerate(conditions):
+            if condition.get('type') == type:
+                continue
+
+            new_conditions.append(condition)
+
+        return new_conditions
+
     def get_condition_idx(self, conditions, name):
```

Comment:
This might be unused now?

Reply:
Yeah, it's unused now, I wasn't sure if there's a need to keep this function around in the toolset.

Reply:
Prob safe to remove